Rspec/spec/solver_spec.rb

```diff
@@ -0,0 +1,30 @@
+require_relative './../../solver'
+
+describe Solver do
+  context 'When testing the Solver class' do
```

Comment:
- [ ] Kindly make sure that if a negative integer is given it should raise an exception.

- Create a method called factorial that takes one argument, an integer N, and returns the [factorial](https://en.wikipedia.org/wiki/Factorial) for that number. The factorial is the multiplication of all integers from 1 to N and has the special case that the factorial of 0 is 1. This method only accepts 0 and positive integers, so if a negative integer is given it should raise an exception.